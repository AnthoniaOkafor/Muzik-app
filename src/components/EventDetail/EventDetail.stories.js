import React from "react";
import { storiesOf } from "@storybook/react";
import { withInfo } from "@storybook/addon-info";
import { withKnobs, text } from "@storybook/addon-knobs";
import EventDetail from "./index";

storiesOf("EventDetail", module)
  .addDecorator(
    withInfo({
      text: "This is a EventDetail component"
    })
  )
  .addDecorator(withKnobs())
  .add("EventDetail", () => (
    <EventDetail 
    about={text("about", "DJ Spinal shutting Down popular club in Yaba")}
    time={text("time", "8:00 PM - Dawn")}
    date={text("date", "28th January, 2020")}/>
  ));