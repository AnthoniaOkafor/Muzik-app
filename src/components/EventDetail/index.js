import React from "react";
import PropTypes from "prop-types";
import "./EventDetail.css";
import Header from "../Header/index";
import Text from "../Text/index";

const EventDetail = ({ about, time, date }) => {
  return (
    <aside className="eventDetails">
      <Header heading="Event Details:" color="#7F8C8D" size={2} />
      <Header heading="About:" color="#7F8C8D" size={3} />
      <Text text={about} size="default" />
      <Header heading="Time:" color="#7F8C8D" size={3} />
      <Text text={time} size="default" />
      <Header heading="Date:" color="#7F8C8D" size={3} />
      <Text text={date} size="default" />
    </aside>
  );
};

EventDetail.defaultProps = {
  about: "Party information",
  time: "20th Jan, 2020",
  date: "date"
};

EventDetail.propTypes = {
  about: PropTypes.string,
  time: PropTypes.string,
  date: PropTypes.string
};
export default EventDetail;
