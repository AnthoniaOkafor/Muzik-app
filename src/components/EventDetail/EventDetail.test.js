import React from "react";
import { mount } from "enzyme";

import EventDetail from "./index";

const about = "DJ Spinal shutting Down popular club in Yaba";
const time = "8:00 PM - Dawn";
const date = "28th January, 2020";

const setUp = (about, time, date) => {
    return mount( <EventDetail 
        about={about}
        time={time}
        date={date}/>);
  };

describe("EventDetail snapshot", () => {
    it("should match snap shot", () => {
      const eventDetail = setUp(about, time, date);
      expect(eventDetail).toMatchSnapshot();
    });
});

// test the number of headers 
describe("renders header", () => {
  it("contains 4 Header component", () => {
    const eventDetail = setUp(about, time, date);
    const header = eventDetail.find("Header");

    expect(header.length).toBe(4);
  });
});

describe("renders Text", () => {

  let eventDetail, paragragh
  beforeEach(() => {
     eventDetail = setUp(about, time, date);
     paragragh = eventDetail.find("p");
  });
  afterEach(() => {
    eventDetail.unmount();
  });

  it("contains all text", () => {
    const header = eventDetail.find("Text");

    expect(header.length).toBe(3);
  });
  it("renders the about correctly", () => {
   
    expect(paragragh.at(0).text()).toEqual(about);
  });
  it("renders the date correctly", () => {

    expect(paragragh.at(1).text()).toEqual(time);
  });
  it("renders the time correctly", () => {

    expect(paragragh.at(2).text()).toEqual(date);
  });
});
