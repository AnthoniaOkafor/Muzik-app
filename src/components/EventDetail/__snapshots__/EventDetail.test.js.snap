// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventDetail snapshot should match snap shot 1`] = `ReactWrapper {}`;
